torch/testing/_internal/common_utils.py

```diff
@@ -817,6 +817,19 @@ class TestCase(expecttest.TestCase):
     # TODO: provide a better mechanism for generated tests to set rtol/atol.
     _precision: float = 0
 
+    # checker to early terminate test suite if unrecoverable failure occurs.
+    def _should_stop_test_suite(self):
+        if torch.cuda.is_initialized():
+            # CUDA device side error will cause subsequence test cases to fail.
+            # stop entire test suite if catches RuntimeError during torch.cuda.synchronize().
+            try:
+                torch.cuda.synchronize()
```

Comment:
Hmm, additional `torch.cuda.synchronize()` after every test would be quite expensive. 
Can you measure the slowdown?
Also, have you checked, if exposing `cudaGetLastError`  into a python runtime would achieve the same but will be much faster?

Reply:
`torch.cuda.synchronize`, if nothing is running on the gpu, is ~1 us, pretty much like any other pytorch operation that was called by the test. cuda tests are expected to do their own synchronization before this call anyway, to test the results.

Reply:
yeah. I actually ran some scuba queries to measure the slowdown (from this PR against he base master commit) 
the result is not significant on cudnn7 tests CI jobs, and if comparing against master commits around that time, it's actually some times faster. 

it would be handy to have @samestep 's test time reporting tool here